File: php/sort/选择排序/selectSort.php
<?php
/*
选择排序
*/ 
 function selectSort(array $arr )
 {
	$len = count($arr); 	
	for($i = 0; $i < $len;$i++){
		 $minIndex = $i; 
		 for($j = $i+1;$j < $len;$j++){
			    if($arr[$j] < $arr[$minIndex] ){
					 $minIndex = $j;
					} 
			 }
		   if($minIndex != $i ){
			     $tmp = $arr[$i];
			     $arr[$i] = $arr[$minIndex]; 
				 $arr[$minIndex] = $tmp;
			   } 
		} 
 	 return $arr;
  } 
   $arr = array(10,56,78,43,49,23,8,3,68);
   $array =  selectSort($arr);
   echo '<pre>';
   print_r($array);

File: C/sort/选择排序/selectSort.c
#include <stdio.h>
#include <stdlib.h>
/*
—°‘Ò≈≈–Ú 
*/
void selectSort(int arr[],int n )
{
	int i = 0,minIndex=0;
	int j,temp;
	 for(i = 0 ; i< n ; i++ ){
 		  minIndex = i;
		 for( j=i+1;j<n;j++){
 		     if(arr[j] < arr[minIndex]){
     		 	minIndex = j;
     		 }	
 		} 
 		if(minIndex != i){
		 	  temp = arr[i];
		 	 arr[i] = arr[minIndex];
		 	 arr[minIndex] = temp;
		 } 
 	}
} 

int main(){
	int arr[] = {10,32,54,665,75,3,5,1,67,13,89};
	int len = sizeof(arr) / sizeof(arr[0]);
	int i = 0;
    selectSort(arr,len);
    for(;i<len;i++){
    	printf("%d\r\n",arr[i]);
    }
     
}
